src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMemberTranslatorPlugin.cs

```diff
@@ -86,8 +86,33 @@ public Expression Translate(MemberExpression e)
                     return new SqlFunctionExpression("ST_NumInteriorRings", typeof(int), new[] { e.Expression });
                 case "NumPoints":
                     return new SqlFunctionExpression("ST_NumPoints", typeof(int),       new[] { e.Expression });
+                case "OgcGeometryType":
+                {
+                    var whenThenList = new List<CaseWhenClause>
+                    {
+                        new CaseWhenClause(Expression.Constant("ST_CircularString"), Expression.Constant(OgcGeometryType.CircularString)),
+                        new CaseWhenClause(Expression.Constant("ST_CompoundCurve"), Expression.Constant(OgcGeometryType.CompoundCurve)),
+                        new CaseWhenClause(Expression.Constant("ST_CurvePolygon"), Expression.Constant(OgcGeometryType.CurvePolygon)),
+                        new CaseWhenClause(Expression.Constant("ST_GeometryCollection"), Expression.Constant(OgcGeometryType.GeometryCollection)),
+                        new CaseWhenClause(Expression.Constant("ST_LineString"), Expression.Constant(OgcGeometryType.LineString)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiCurve"), Expression.Constant(OgcGeometryType.MultiCurve)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiLineString"), Expression.Constant(OgcGeometryType.MultiLineString)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiPoint"), Expression.Constant(OgcGeometryType.MultiPoint)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiPolygon"), Expression.Constant(OgcGeometryType.MultiPolygon)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiSurface"), Expression.Constant(OgcGeometryType.MultiSurface)),
+                        new CaseWhenClause(Expression.Constant("ST_Point"), Expression.Constant(OgcGeometryType.Point)),
+                        new CaseWhenClause(Expression.Constant("ST_Polygon"), Expression.Constant(OgcGeometryType.Polygon)),
+                        new CaseWhenClause(Expression.Constant("ST_PolyhedralSurface"), Expression.Constant(OgcGeometryType.PolyhedralSurface)),
+                        new CaseWhenClause(Expression.Constant("ST_Tin"), Expression.Constant(OgcGeometryType.TIN))
+                    };
+
+                    return new CaseExpression(
+                        new SqlFunctionExpression("ST_GeometryType", typeof(string), new[] { e.Expression }),
+                        whenThenList.ToArray());
+                }
                 case "PointOnSurface":
-                    return new SqlFunctionExpression("ST_PointOnSurface", typeof(IPoint), new[] { e.Expression });
+                case "InteriorPoint":
```

Comment:
Are these two functions actually identical? I'm far from a spatial expert but the names make it sound like `PointOnSurface` could be a point on the edge, whereas `InteriorPoint` can't.

Reply:
I agree that the descriptions point to that, but in the code, `PointOnSurface` actually just maps to `InteriorPoint`: https://github.com/NetTopologySuite/NetTopologySuite/blob/develop/NetTopologySuite/Geometries/Geometry.cs#L501

Reply:
Thanks for the pointer. I'm still wondering whether this is the required standard behavior or not (i.e. is NTS doing the right thing), but I guess it's OK.

Reply:
I'm trying to find some sort of official documentation for `InteriorPoint` but it doesn't seem to be part of any standard. Happy to change it if someone can point me in the right direction, or remove it entirely.

Reply:
Nah, it's OK. It may have been an initiative of the NetTopologySuite folk rather than part of any spatial standard. If you wish to, you can check with them - but I'm also OK with going ahead as-is.